### PR TITLE
Document microflow URLs and update page URL documentation

### PR DESCRIPTION
**Pull request by WimJongeneel**
Planned for 10.3

### CODE REVIEW

File: content/en/docs/refguide/modeling/pages/data-widgets/data-view.md [hunk 1]
```diff
@@ -77,7 +77,7 @@ The data view supports the following types of data source: context, microflow, n
...
+| Context          | A data source that uses a context object which is already available, such as a page parameter, snippet parameter or the object of a surrounding data container. Page parameters are passed to the page from another document, such as an [Show Page action](/refguide/on-click-event/#show-page) in another page or a [Show Page activity](/refguide/show-page/) in a microflow or nanoflow. In the case of a snippet parameter the data will come from a [Snippet Call](/refguide/snippet-call/) in a [page](/refguide/page/), [layout](/refguide/layout/), or another [snippet](/refguide/snippet/). If the context object comes from a data container, you must specify an entity path that starts in the context object and follows one or more associations. For more information on the context source, see [Context Source](/refguide/context-source/). |
```
Comment (ConnorLand) [inline content/en/docs/refguide/modeling/pages/data-widgets/data-view.md:80]: I need to check this change, but it's a discrepancy I found while editing this PR.

File: content/en/docs/refguide/modeling/application-logic/microflows-and-nanoflows/microflows/microflow.md [hunk 2]
```diff
@@ -29,6 +29,29 @@ Microflow properties consist of the following sections:
 
 **Name** is the internal name of the microflow. When referring to the microflow in the app you will use this name. It must be unique within the module, but you can have two microflows with the same name in different modules. When referring to the microflow, you will normally prepend the name of the module to ensure uniqueness and allow you to use microflows in other modules.
 
+#### 2.1.2 URL
+
+Microflow URLs allow you to execute a microflow when the user navigates to a specific URL within your application. The microflow will be executed during the client startup, just before the home page is shown. When the microflow executes a [Show page](/refguide/on-click-event/#show-page) action. The microflow's full URL will be the base URL of your application, followed by `/p/`, followed by the page's configured URL.
```
Comment (Robinfr) [inline content/en/docs/refguide/modeling/application-logic/microflows-and-nanoflows/microflows/microflow.md]: > When the microflow executes a [Show page](/refguide/on-click-event/#show-page) action.

What? I'm missing half a sentence I think

Comment (Robinfr) [inline content/en/docs/refguide/modeling/application-logic/microflows-and-nanoflows/microflows/microflow.md]: > The microflow's full URL will be the base URL of your application, followed by `/p/`, followed by the page's configured URL.

I think you mean microflow's configured URL?

File: content/en/docs/refguide/modeling/application-logic/microflows-and-nanoflows/microflows/microflow.md [hunk 3]
```diff
@@ -29,6 +29,29 @@ Microflow properties consist of the following sections:
 
 **Name** is the internal name of the microflow. When referring to the microflow in the app you will use this name. It must be unique within the module, but you can have two microflows with the same name in different modules. When referring to the microflow, you will normally prepend the name of the module to ensure uniqueness and allow you to use microflows in other modules.
 
+#### 2.1.2 URL
+
+Microflow URLs allow you to execute a microflow when the user navigates to a specific URL within your application. The microflow will be executed during the client startup, just before the home page is shown. When the microflow executes a [Show page](/refguide/on-click-event/#show-page) action. The microflow's full URL will be the base URL of your application, followed by `/p/`, followed by the page's configured URL. 
+
+Take this URL for an example: `http://example.mendixcloud.com/p/microflow`.
```
Comment (Robinfr) [inline content/en/docs/refguide/modeling/application-logic/microflows-and-nanoflows/microflows/microflow.md]: Usually, if you give an example, you're going to say something about the example. But here, we just have an example, and then.. nothing? 

I think what you were trying to do is give an example for the previous paragraph, in that case you can change this to something like:

```suggestion
For example: `http://example.mendixcloud.com/p/microflow`.
```

But it should be part of the previous paragraph (at the end).

File: content/en/docs/refguide/modeling/application-logic/microflows-and-nanoflows/microflows/microflow.md [hunk 4]
```diff
@@ -29,6 +29,29 @@ Microflow properties consist of the following sections:
...
+
+For object parameters, the name of the parameter and the name of the attribute is used, for example `my-microflow/{Product/Name}`. Primitive parameters will be parsed from the URL and their values directly passed to the microflow. However, for object parameters an XPath query is used to retrieve the object by the value of the attribute.
+
+For example, in the URL `product/{Product/Name}` the `Name` attribute of the parameter `Product` will be used in the URL (in a browser, the URL appears as `http://example.mendixcloud.com/p/product/hammer`). Any attribute or primitive parameter of types `Boolean`, `Decimal`, `Enumeration`, `Integer`, `Long`, or `String` can be used in the URL. 
+
+Next to this, you can use `Id` as an attribute to include the entity's identifier in the URL. This would appear as `product/{Product/Id}` for example. 
+
+In the microflow URL dialog box, the configured URL is shown together with an example URL with example values filled in for the parameters. It and also shows how the parameter will be retrieved:
```
Comment (Robinfr) [inline content/en/docs/refguide/modeling/application-logic/microflows-and-nanoflows/microflows/microflow.md]:
```suggestion
In the microflow URL dialog box, the configured URL is shown together with an example URL with example values filled in for the parameters. It also shows how the parameter will be retrieved:
```

File: content/en/docs/refguide/modeling/application-logic/microflows-and-nanoflows/microflows/microflow.md [hunk 5]
```diff
@@ -29,6 +29,29 @@ Microflow properties consist of the following sections:
...
+
+{{<figure src="/attachments/refguide/modeling/application-logic/microflows-and-nanoflows/microflows/microflow/microflow-url.jpg" alt="microflow url dialog" >}}
+
+
+{{% alert color="warning" %}}
+URLs are not supported for microflows that have non-persistable entities or lists as parameters.
```
Comment (Robinfr) [inline content/en/docs/refguide/modeling/application-logic/microflows-and-nanoflows/microflows/microflow.md:49]: Are there any types of parameters/attributes we don't support? Such as date times?

**Review by Robinfr — CHANGES_REQUESTED** [review]

File: content/en/docs/refguide/modeling/application-logic/microflows-and-nanoflows/microflows/microflow.md (hunk 5 above)
Comment (WimJongeneel) [reply content/en/docs/refguide/modeling/application-logic/microflows-and-nanoflows/microflows/microflow.md:49]: We do mention the supported types above:

> Any attribute or primitive parameter of types `Boolean`, `Decimal`, `Enumeration`, `Integer`, `Long`, or `String` can be used in the URL.

But we could also add it to the limitations here, but then we should also do the same for the Page URL documentation.

Comment (Robinfr) [reply content/en/docs/refguide/modeling/application-logic/microflows-and-nanoflows/microflows/microflow.md:49]: Ah if it's mentioned already it's fine.

**Review by Robinfr — APPROVED** [review]

**Comment by Robinfr** [comment]
@ConnorLand Can be reviewed by you :) (Though I think you already started 😄)

**Comment by Yiyun333** [comment]
@ConnorLand I've finished my review on the microflow URL section. Please continue with your review.

**Comment by Yiyun333** [comment]
> @ConnorLand I've finished my review on the microflow URL section. Please continue with your review.

Sorry! I had to fix an anchor error in the build. Now there is no error in the build. Please continue :)

**Review by ConnorLand — APPROVED** [review]

**Pull request merged**